user/mongoStoreClient.go

```diff
@@ -96,6 +97,10 @@ func (d MongoStoreClient) FindUsers(user *User) (results []*User, err error) {
 		fieldsToMatch = append(fieldsToMatch, bson.M{"emails": bson.M{"$in": user.Emails}})
 	}
 
+	if len(user.Roles) > 0 {
+		fieldsToMatch = append(fieldsToMatch, bson.M{"roles": bson.M{"$in": user.Roles}})
```

Comment:
This will not work in a case sensitive manner.  I'd suggest we force all roles to be lowercase at all points (command-line tool and API).